src/mozilla_sec_eia/models/sec_eia_record_linkage/transform_sec_input.py

```diff
@@ -0,0 +1,371 @@
+
+
+@multi_asset(
+    ins={
+        "ex21_df": AssetIn("ex21_company_ownership_info"),
+    },
+    outs={
+        "transformed_ex21_subsidiary_table": AssetOut(
+            io_manager_key="pandas_parquet_io_manager",
+        )
+    },
+    partitions_def=year_quarter_partitions,
+)
+def transform_ex21_table(
+    ex21_df: pd.DataFrame, sec10k_filing_metadata: pd.DataFrame
+) -> pd.DataFrame:
+    """Transform Ex. 21 table of subsidiaries before combining with basic 10k table."""
+    ex21_df.loc[:, "filename"] = convert_ex21_id_to_filename(ex21_df)
+    ex21_df = ex21_df.drop(columns=["id"])
+    ex21_df = _add_report_year_to_sec(ex21_df, sec10k_filing_metadata)
+    ex21_df = ex21_df.rename(columns=EX21_COL_MAP)
+    ex21_df = clean_location_of_inc(ex21_df)
+    # TODO: what to do with the clean company name?
+    ex21_df = transform_company_name(ex21_df)
+    ex21_df = _add_parent_company_cik(ex21_df, sec10k_filing_metadata)
+    # add an sec_company_id, ultimately this ID become the subsidiary's CIK
+    # if the subsidiary is matched to an SEC filer
+    ex21_df = create_sec_company_id_for_ex21_subs(ex21_df=ex21_df)
+    ex21_df = _flatten_sec_companies_across_time(ex21_df)
+    ex21_df = ex21_df.fillna(np.nan)
+
+    return ex21_df
+
+
+def transform_basic10k_table(
+    basic_10k_df: pd.DataFrame, sec10k_filing_metadata: pd.DataFrame
+) -> pd.DataFrame:
+    """Transformations on SEC basic 10K filer table to prepare for record linkage."""
+    basic_10k_df = basic_10k_df.reset_index().pivot_table(
+        values="value", index="filename", columns="key", aggfunc="first"
+    )
+    basic_10k_df.columns.name = None
+    # TODO: chain these function calls together
+    basic_10k_df = basic_10k_df.reset_index()
+    basic_10k_df = _remove_weird_sec_cols(basic_10k_df)
+    basic_10k_df = _add_report_year_to_sec(basic_10k_df, sec10k_filing_metadata)
+    basic_10k_df = basic_10k_df.rename(columns=SEC_COL_MAP)
+    # add a location of incorporation to better match it to Ex. 21 subsidiaries
+    basic_10k_df = clean_location_of_inc(basic_10k_df)
+    basic_10k_df = transform_company_name(basic_10k_df)
+    basic_10k_df.loc[:, "zip_code"] = basic_10k_df["zip_code"].str[:5]
+    basic_10k_df = fill_street_address_nulls(basic_10k_df)
+    basic_10k_df.loc[:, "files_10k"] = True
+    basic_10k_df.loc[:, "sec_company_id"] = basic_10k_df["central_index_key"]
+    basic_10k_df[STR_COLS] = basic_10k_df[STR_COLS].apply(
+        lambda x: x.str.strip().str.lower()
+    )
+    return basic_10k_df
+
+
+@multi_asset(
+    ins={
+        "basic_10k_df": AssetIn("basic_10k_company_info"),
+        "clean_ex21_df": AssetIn("transformed_ex21_subsidiary_table"),
+        # specify an io_manager_key?
+    },
+    outs={
+        "core_sec_10k__parents_and_subsidiaries": AssetOut(
+            io_manager_key="pandas_parquet_io_manager",
+            # specify a dagster_type?
+        ),
+    },
+    partitions_def=year_quarter_partitions,
+)
```

Comment:
Hmm I'm not sure if `multi_asset` is the right thing to use for this table. I'm not sure if the Ex. 21 cleaned subsidiaries table is useful by itself, maybe it should just be an upstream operation (or asset) from the SEC output table.